apps/memcached/tests/test_memcached.py

```diff
@@ -257,7 +257,7 @@ def test_expiry(self):
 
     @slow
     def test_expiry_at_epoch_time(self):
-        expiry = int(time.time()) + 1
+        expiry = int(time.time()) + 2
         self.assertEqual(call('set key 0 %d 5\r\nhello\r\n' % expiry), b'STORED\r\n')
         self.assertEqual(call('get key\r\n'), b'VALUE key 0 5\r\nhello\r\nEND\r\n')
         time.sleep(2)
```

Comment:
Don't you need to increase this sleep as well to compensate for the larger expiry?

Reply:
@tgrabiec indeed. fixed and repushed.

Reply:
I think that if sleep(2) guarantees that it sleeps **at least** 2 seconds (never less), then it was fine even after the  +2 expiry. But better safe than sorry, I guess.

Reply:
it could sleep less than 2 second if we take the RTT time and the worse-case time drift into consideration.

Reply:
@tchaikov no, on Linux a sleep(2) is guaranteed to sleep at least 2 seconds - not less. And there is no "time drift" in the tests (everything is running on the same machine), and I don't understand why RTT is relevant.

Any any case, I don't see any harm in increasing the sleep. In test/alternator I view a sleep(3) as a disaster, but this sort of anti-sleeping isn't common in our other tests so we don't have to start here.

Reply:
ahh, right. seastar uses a monotonical timer, so sleep for 2 second should always work.

Reply:
by time drift, i was referencing a scenario:

1. the wall clock is 1 second off
2. ntp fixes it by adjust the wall clock

but this only changes the wall clock.

Reply:
> ahh, right. seastar uses a monotonic timer, so sleep for 2 second should always work.

You're right that clock changes can break this test, but changing 2 to 3 doesn't change that: 
The sleep(2) will always sleep two seconds, but if the clock changed considerably during that time, memcached might think that the current time is still less than the expiration time, and not expire the item.

I guess your adding of an extra second can help in case that ntp decides to change the clock and this change is less than one second during the two second sleep. If ntpd uses `adjtime()` (as it should), this will indeed happen. If ntp notices some huge correction and does it immediately, it can still break the test, but I think we don't need the test to survive such an unlikely situation.

Reply:
> I think that if sleep(2) guarantees that it sleeps at least 2 seconds (never less), then it was fine even after the +2 expiry. But better safe than sorry, I guess.

if expiry is set to `int(now()) + 2`, then in the worst case the item can be expired at now() + 3 because the expiry is done once per second periodically every second. If now() is 3.0000001 and expiry timer fires at 3.9999999, it will expire the item at 5.9999999